### ISSUE TITLE
ECDSA not supported?

### ISSUE THREAD

**Issue by sgammon** [grpc/grpc-swift#150]
After debugging some connection failures from `grpc-swift` to my gRPC services running on TLS, I found that handshakes were failing because gRPC Swift only advertises RSA ciphers.

After initializing a service with:
```swift
  let _ = gRPC_Service.init(address: "[redacted]:443", secure: true)
```

I get the following packet capture for the `CilentHello`:
![screen shot 2018-02-27 at 2 33 10 pm](https://user-images.githubusercontent.com/171897/36759176-6188b7ce-1bcb-11e8-9251-d751a38fa1dc.png)

We usually fly in prod with ECDSA keys, so the handshake fails. I know `grpc-swift` uses BoringSSL under the hood, but I don't see a way to control cipher selection.

Is it possible to support ECDSA, and if so, why is it not supported? Thank you in advance for your help

**Comment by timburks** [comment]
Hi, thanks for raising this; I wasn't aware of it. @muxi could you comment on support for ECDSA in the gRPC C core?

**Comment by dcow** [comment]
@sgammon try setting `GRPC_SSL_CIPHER_SUITES` in your environment.

https://github.com/grpc/grpc/blob/dc46c7f565196d6684d5cf6ee81e99b413ba5282/src/core/lib/security/security_connector/security_connector.cc#L80

**Comment by sgammon** [comment]
@timburks and @dcow, thank you for the replies. i'll give this a shot in my environment. knowing where to override in cgRPC helps too.

shouldn't ECDSA suites be enabled by default, though? especially for mobile or CPU-constrained environments, where I imagine gRPC Swift will run rather frequently, ECC-based keying can offer significant benefits.

i'll try raising a PR and see what the gRPC core team says.

**Comment by sgammon** [comment]
hey all - i've filed a PR with gRPC core, at grpc/grpc#14656. I'll be monitoring that PR for feedback etc

**Comment by sgammon** [comment]
update: looks like this was fixed in core at grpc/grpc#15404
